### DISCUSSION TITLE
[Bug]: Virtualized Select still does render all options on the DOM

### DISCUSSION THREAD

**Discussion by xeinebiu** [chakra-ui/ark#3022] [General]
### Description

The issue is reproducable even on the demostration of Virtualized Select. You can see that all countries <option> elements are rendered on the DOM which defeats the virtualization purpose.

![Screenshot 2024-10-28 200822](https://github.com/user-attachments/assets/a0b73d1f-9973-465b-88b8-335209e3008f)
![image](https://github.com/user-attachments/assets/f7d8adea-b424-499e-8ea9-69059a033828)

### Link to Reproduction (or Detailed Explanation)
https://ark-ui.com/react/examples/virtualized-select

### Ark UI Version
4.2.0

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

### Additional Information

Solution
> Do not include the  <Select.HiddenSelect />, I am not sure what is the purpose of HiddenSelect but removing this fixes the issue.

**Comment by cschroeter** [comment]
@xeinebiu

Thanks for taking the time to create this issue! Yes, avoiding `HiddenSelect` would prevent hidden options in the DOM. However, for this example with ~250 items, it should work fine.

If you plan to exceed 10k items, we definitely recommend **not** using a Hidden Select.

**Reply by xeinebiu** [reply]
@cschroeter thanks for your response. May I ask, whats the role of HiddenSelect?

**Reply by cschroeter** [reply]
@xeinebiu Its primary use case is for form submissions.

```tsx
const { register, handleSubmit } = useForm<Inputs>();

<Select.HiddenSelect {...register('framework')} />;
```

Alternatively, you can replace `Select.HiddenSelect` with a `Controller`. Here's an example using `react-hook-form`:  
https://react-hook-form.com/get-started#IntegratingControlledInputs

**Comment by segunadebayo** [comment]
For virtualized lists with many items, skip `Select.HiddenSelect` and create a lightweight hidden input that only submits the selected value:

```tsx
import { Select, useSelectContext } from '@ark-ui/react/select'

const SelectHiddenInput = ({ name }: { name: string }) => {
  const select = useSelectContext()
  return (
    <>
      {select.value.map((value) => (
        <input key={value} type="hidden" name={name} value={value} />
      ))}
    </>
  )
}
```

Then use it in place of `Select.HiddenSelect`:

```tsx
<Select.Root collection={collection}>
  <Select.Control>
    <Select.Trigger>
      <Select.ValueText placeholder="Select country" />
    </Select.Trigger>
  </Select.Control>
  <Select.Positioner>
    <Select.Content>{/* virtualized items */}</Select.Content>
  </Select.Positioner>
  <SelectHiddenInput name="country" />
</Select.Root>
```

This avoids rendering thousands of `<option>` elements in the DOM while still working with native form submission.
